dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/model/AbstractResourceInfo.java

```diff
@@ -154,39 +239,48 @@ public void setResourceClass(Class<?> rClass) {
+    private void findContextFields(final Class<?> cls, Object provider, ProviderContextInfo contextInfo) {
         if (cls == Object.class || cls == null) {
             return;
         }
+
         //Liberty code change start
-        Field[] fields = AccessController
-                        .doPrivileged(new PrivilegedAction<Field[]>() {
-                            @Override
-                            public Field[] run() {
-                                return cls.getDeclaredFields();
+        if (contextInfo == null || contextInfo.fieldNames != null) {
+            Field[] fields = AccessController.doPrivileged(new PrivilegedAction<Field[]>() {
+                @Override
+                public Field[] run() {
+                    return cls.getDeclaredFields();
+                }
+            });
+            for (Field f : fields) {
+                //Liberty code change end
+                for (Annotation a : f.getAnnotations()) {
+                    if (a.annotationType() == Context.class) {
+                        contextFields = addContextField(contextFields, f);
+                        if (f.getType().isInterface()) {
+                            checkContextClass(f.getType());
+                            //Liberty code change start defect 169218
+                            //Add the FieldProxy to the set
+                            ThreadLocalProxy<?> proxy = getFieldThreadLocalProxy(f, provider);
+                            boolean added = addToMap(getFieldProxyMap(true), f, proxy);
+                            if (added) {
+                                ThreadLocalProxyCopyOnWriteArraySet<ThreadLocalProxy<?>> proxySet = getProxySet();
+                                proxySet.add(proxy);
                             }
-                        });
-        for (Field f : fields) {
-            //Liberty code change end
-            for (Annotation a : f.getAnnotations()) {
-                if (a.annotationType() == Context.class) {
-                    contextFields = addContextField(contextFields, f);
-                    if (f.getType().isInterface()) {
-                        checkContextClass(f.getType());
-                        //Liberty code change start defect 169218
-                        //Add the FieldProxy to the set
-                        ThreadLocalProxy<?> proxy = getFieldThreadLocalProxy(f, provider);
-                        boolean added = addToMap(getFieldProxyMap(true), f, proxy);
-                        if (added) {
-                            ThreadLocalProxyCopyOnWriteArraySet<ThreadLocalProxy<?>> proxySet = getProxySet();
-                            proxySet.add(proxy);
+                            //Liberty code change end
                         }
-                        //Liberty code change end
                     }
                 }
             }
         }
-        findContextFields(cls.getSuperclass(), provider);
+
```

Comment:
can you surround this code in `//Liberty change start/end` comments?

Reply:
done